semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -376,14 +401,15 @@ def _populate_extends_single(
                     semconv.constraints += (constraint.inherit_anyof(),)
             # Attributes
             parent_attributes = {}
-            for ext_attr in extended.attrs_by_name.values():
+            for ext_attr in extended.attributes:
                 parent_attributes[ext_attr.fqn] = ext_attr.inherit_attribute()
             # By induction, parent semconv is already correctly sorted
             parent_attributes.update(
                 SemanticConventionSet._sort_attributes_dict(semconv.attrs_by_name)
             )
-            semconv.attrs_by_name = parent_attributes
-        else:  # No parent, sort of current attributes
+            if parent_attributes or semconv.attributes:
+                semconv.attrs_by_name = parent_attributes
+        elif semconv.attributes:  # No parent, sort of current attributes
```

Comment:
This is added so we only try to sort attributes for semantic conventions that actually have attributes (spans and resources).